### PR TITLE
Update setup.py to install sumo as a package

### PR DESCRIPTION
**Pull request by RishabhJain2018**
Steps to install it locally:
1. Clone the repository and switch to this PR
2. Go to `sumo-api` directory
3. Run `pip install -r requirements.txt`
4. Run `pip install .`

The sumo package will be installed locally.

To Verify it:
1. Open python shell
2. Run `import sumo`
The command should run properly without any error.

### CODE REVIEW

File: sumo-api/setup.py [hunk 1]
```diff
@@ -140,5 +142,17 @@
...
+    ext_modules = cythonize(extensions),
+    name='sumo',
+    version='1.0',
+    description='Sumo challenge python package',
+    install_requires=[
+      'cython',
+      'easydict',
+      'matplotlib',
+      'numpy',
+      'scipy'
```
[the remark this answers is not in the document]
Comment (RishabhJain2018) [reply sumo-api/setup.py]: Yeah, I agree. Most of the famous packages like `requests` also do it the same way. I'll update it.

**Comment by RishabhJain2018** [comment]
I have updated the `setup.py` file but there are two dependencies `Cython` and `numpy` on which the execution of `setup.py` depends. Hence, I wasn't able to remove them from the `requirements.txt` file.

**Comment by dhuber23b** [comment]
I don't know why this is happening, but I get the following error when I try the setup:

(venv) dhuber@ubuntu:~/sumo-challenge/sumo-api$ python setup.py build_ext --inplace
Compiling sumo/base/colored_category.pyx because it changed.
Compiling sumo/base/vector.pyx because it changed.
Compiling sumo/geometry/inverse_depth.pyx because it changed.
Compiling sumo/geometry/pose3.pyx because it changed.
Compiling sumo/geometry/quaternion.pyx because it changed.
Compiling sumo/geometry/rot3.pyx because it changed.
Compiling sumo/opencv/wrap.pyx because it changed.
Compiling sumo/images/multi_image_tiff.pyx because it changed.
Compiling sumo/images/rgbd_360.pyx because it changed.
Compiling sumo/images/rgbd_tiff.pyx because it changed.
Compiling sumo/images/rgbdci_360.pyx because it changed.
Compiling sumo/threedee/box_3d.pyx because it changed.
Compiling sumo/threedee/compute_bbox.pyx because it changed.
Compiling sumo/threedee/glb_converter.pyx because it changed.
Compiling sumo/threedee/gltf.pyx because it changed.
Compiling sumo/threedee/gltf_model.pyx because it changed.
Compiling sumo/threedee/mesh.pyx because it changed.
Compiling sumo/threedee/model_sampler.pyx because it changed.
Compiling sumo/threedee/point_cloud.pyx because it changed.
Compiling sumo/threedee/point_cloud_ply.pyx because it changed.
Compiling sumo/threedee/textured_mesh.pyx because it changed.
Compiling sumo/threedee/voxelizer.pyx because it changed.
[ 1/22] Cythonizing sumo/base/colored_category.pyx
[ 2/22] Cythonizing sumo/base/vector.pyx
[ 3/22] Cythonizing sumo/geometry/inverse_depth.pyx
[ 4/22] Cythonizing sumo/geometry/pose3.pyx
[ 5/22] Cythonizing sumo/geometry/quaternion.pyx
[ 6/22] Cythonizing sumo/geometry/rot3.pyx
[ 7/22] Cythonizing sumo/images/multi_image_tiff.pyx
[ 8/22] Cythonizing sumo/images/rgbd_360.pyx
[ 9/22] Cythonizing sumo/images/rgbd_tiff.pyx
[10/22] Cythonizing sumo/images/rgbdci_360.pyx
[11/22] Cythonizing sumo/opencv/wrap.pyx
[12/22] Cythonizing sumo/threedee/box_3d.pyx
[13/22] Cythonizing sumo/threedee/compute_bbox.pyx
[14/22] Cythonizing sumo/threedee/glb_converter.pyx
[15/22] Cythonizing sumo/threedee/gltf.pyx
[16/22] Cythonizing sumo/threedee/gltf_model.pyx
[17/22] Cythonizing sumo/threedee/mesh.pyx
[18/22] Cythonizing sumo/threedee/model_sampler.pyx
[19/22] Cythonizing sumo/threedee/point_cloud.pyx
[20/22] Cythonizing sumo/threedee/point_cloud_ply.pyx
[21/22] Cythonizing sumo/threedee/textured_mesh.pyx
[22/22] Cythonizing sumo/threedee/voxelizer.pyx
running build_ext
error: each element of 'ext_modules' option must be an Extension instance or 2-tuple

**Comment by dhuber23b** [comment]
Note: I checked and each element of ext_modules is an Extension instance, so I really don't understand the error message.  I am going to try to do a clean build and see if I still get the error message.

**Comment by RishabhJain2018** [comment]
@dhuber23b I have updated the PR, please check. It should work now.

**Comment by dhuber23b** [comment]
It works!  Thanks for the fix.

**Pull request merged**